arwen/elrondapi/bigintOps.go

```diff
@@ -590,9 +592,10 @@ func bigIntEDiv(context unsafe.Pointer, destination, op1, op2 int32) {
 	dest, a, b := bigInt.GetThree(destination, op1, op2)
 	if b.Sign() == 0 {
 		runtime := arwen.GetRuntimeContext(context)
-		runtime.SignalUserError("[bigIntEDiv] division by 0")
+		runtime.SignalUserError(arwen.UserErrorDivZero)
+	} else {
+		dest.Div(a, b) // Div implements Euclidean division (unlike Go)
 	}
-	dest.Div(a, b) // Div implements Euclidean division (unlike Go)
 
 	gasToUse := metering.GasSchedule().BigIntAPICost.BigIntEDiv
```

Comment:
Move gas consumption immediately after retrieving the `bigInt` and `metering` contexts, for all the functions of the Big Int API, to ensure gas is consumed in case of error.

Reply:
Good idea, done.